tests/agent_server/deploy/e2e/test_autostarted.py

```diff
@@ -84,3 +88,159 @@ async def test_auto_deploy_no_splay(server, client, clienthelper, resource_conta
+    assert (
+        len(beats) < duration * 10
+    ), f"Sent {len(beats)} heartbeats over a time period of {duration} seconds, sleep mechanism is broken"
+
+
+@pytest.mark.parametrize(
+    "agent_repair_interval",
+    [
+        "2",
+        "*/2 * * * * * *",
+    ],
+)
+async def test_spontaneous_repair(server, client, agent, resource_container, environment, clienthelper, agent_repair_interval):
+    """
+    Test that a repair run is executed every 2 seconds in the new agent
+     as specified in the agent_repair_interval (using a cron or not)
+    """
+    resource_container.Provider.reset()
+    env_id = environment
+
+    Config.set("config", "agent-repair-interval", agent_repair_interval)
+    Config.set("config", "agent-repair-splay-time", "2")
+    Config.set("config", "agent-deploy-interval", "0")
+
+    # This is just so we can reuse the agent from the fixtures with the new config options
+    agent._set_deploy_and_repair_intervals()
+    agent._enable_time_triggers()
+    version = await clienthelper.get_version()
+
+    resources = [
+        {
+            "key": "key1",
+            "value": "value1",
+            "id": "test::Resource[agent1,key=key1],v=%d" % version,
+            "purged": False,
+            "send_event": False,
+            "requires": [],
+        },
+    ]
+
+    result = await client.put_version(
+        tid=env_id, version=version, resources=resources, unknowns=[], version_info={}, compiler_version=get_compiler_version()
+    )
+    assert result.code == 200
+
+    # do a deploy
+    result = await client.release_version(env_id, version, True, const.AgentTriggerMethod.push_full_deploy)
+    assert result.code == 200
+    assert not result.result["model"]["deployed"]
+    assert result.result["model"]["released"]
+    assert result.result["model"]["total"] == 1
+    assert result.result["model"]["result"] == "deploying"
+
+    result = await client.get_version(env_id, version)
+    assert result.code == 200
+
+    await clienthelper.wait_full_success(env_id)
+
+    async def verify_deployment_result():
+        result = await client.get_version(env_id, version)
+        # A repair run may put one resource from the deployed state to the deploying state.
+        assert len(resources) - 1 <= result.result["model"]["done"] <= len(resources)
+
+        assert resource_container.Provider.isset("agent1", "key1")
+        assert resource_container.Provider.get("agent1", "key1") == "value1"
+
+    await verify_deployment_result()
+
+    # Manual change
+    resource_container.Provider.set("agent1", "key1", "another_value")
+    # Wait until repair restores the state
+    now = time.time()
+    while resource_container.Provider.get("agent1", "key1") != "value1":
+        if time.time() > now + 10:
+            raise Exception("Timeout occurred while waiting for repair run")
+        await asyncio.sleep(0.1)
```

Comment:
The [`retry_limited`](https://github.com/inmanta/inmanta-core/blob/c4913c81e47e6dd01aaaa64a4372085d0dcf962b/tests/utils.py#L68) method can be used for this pattern.